### ISSUE TITLE
Class members are inaccessible in `tryLock` 

### ISSUE THREAD

**Issue by nerdDan** [miaowing/nest-schedule#5]
Class members are inaccessible due to the class not being bound in the execution context of `tryLock`.

Following snippet explains what I mean.
```typescript
import { Cron, NestDistributedSchedule } from 'nest-schedule';

export class Foo extends NestDistributedSchedule {
  public lock: boolean;

  constructor() {
    super();
    this.lock = false;
  }

  public async tryLock(method: string) {
    if (this.lock) { // runtime error, because `this` is undefined
      throw new Error();
    }
    this.lock = true;
    return () => {
      this.lock = false;
    };
  }

  @Cron('*/1 * * * * *')
  public async cron(): Promise<void> {
    console.log(new Date());
  }
}

const foo = new Foo();
```

Perhaps `.bind` should be added to
https://github.com/miaowing/nest-schedule/blob/master/lib/JobExecutor.ts#L19

**Comment by amirsaber** [comment]
I am trying to implement an advisory lock using postgresql and typeorm. I am facing same issue when trying to use the connection that is injected into the constructor.

**Comment by miaowing** [comment]
Thanks, I have fixed it in 0.4.1, @nerdDan @amirsaber
